lib/aasm/persistence/base.rb

```diff
@@ -84,6 +84,11 @@ def state_with_scope(name, *args)
     end
     alias_method :state_without_scope, :state
     alias_method :state, :state_with_scope
+
+  private
+    def should_create_scope?(name, *args)
+      return @state_machine.config.create_scopes && !@klass.respond_to?(name) && (args.first.blank? || args.first[:create_scope] != false)
```

Comment:
Heh, me neither to be honest. The only thing which came to my mind is to fix scenarios like this one but still - not the best solution as you wrote.